File: problem3.java
import java.util.Queue;
import java.util.LinkedList;

class Solution {
    public int countStudents(int[] students, int[] sandwiches) {
        Queue<Integer> studentQueue = new LinkedList<>();

        for (int value : students) {
            studentQueue.add(value);
        }

        int sandwichIndex = 0;
        int unableToEat = 0;
        while (!studentQueue.isEmpty() && unableToEat < studentQueue.size()) {
            if (sandwiches[sandwichIndex] == studentQueue.peek()) {
                unableToEat = 0;

                ++sandwichIndex;
                studentQueue.poll();
            } else {
                ++unableToEat;

                studentQueue.add(studentQueue.poll());
            }
        }

        return unableToEat;
    }
}

File: problem3.py
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        
        n =len(sandwiches)
        i = 0
        while i < (len(students)):
            v = students[0]
            if v == sandwiches[0]:
                sandwiches.pop(0)
                students.pop(0)
                i = 0 
            else:
                k = students.pop(0)
                students.append(k)
                i += 1
        return len(students)
